File: platforms/ios/Packages/FunputKit/Sources/FunputShared/Shortcuts/TextShortcut.swift
import Foundation

/// A user-authored text replacement. Its identity survives edits and reloads.
public struct TextShortcut: Codable, Hashable, Sendable, Identifiable {
    public let id: UUID
    public var trigger: String
    public var expansion: String

    public init(id: UUID = UUID(), trigger: String = "", expansion: String = "") {
        self.id = id
        self.trigger = trigger
        self.expansion = expansion
    }

    public var isValid: Bool {
        !trigger.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
            && !expansion.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }
}

File: platforms/android/shortcut-store/src/main/java/app/funput/funput/shortcuts/model/TextShortcut.kt
package app.funput.funput.shortcuts.model

import java.util.UUID

/** A user-authored text replacement whose identity survives edits and reloads. */
data class TextShortcut(
    val id: UUID = UUID.randomUUID(),
    val trigger: String = "",
    val expansion: String = "",
) {
    val isValid: Boolean
        get() = trigger.isNotBlank() && expansion.isNotBlank()
}
